prody/proteins/interactions.py

```diff
@@ -761,31 +761,29 @@ def calcDisulfideBonds(atoms, distA=2.5):
                             'with `getCoords` method')
 
     try:
-        atoms_SG = atoms.select('protein and resname CYS')
-    except AttributeError:
-        try:
-            checkCoords(atoms_SG)
-        except TypeError:
-            raise TypeError('Lack of cysteines in the structure.')
+        atoms_SG = atoms.select('protein and resname CYS and name SG')
```

Comment:
You could make the selection outside a try/except and then use an if to check if there are cys sg atoms.

Reply:
I modified this function a few times with different try/except etc. I think I also did the way you described, but after tests on different structures, I had to modify it. For some reason, it was working badly on one structure that I tested. The code which we have now was working correctly on all structures it was tested.

Reply:
ok